include/kindr/math/LinearAlgebra.hpp

```diff
@@ -80,7 +80,7 @@ bool static pseudoInverse(const _Matrix_TypeA_ &a, _Matrix_TypeB_ &result,
   // If one dimension is dynamic, compute everything as dynamic size
   constexpr auto m = Eigen::JacobiSVD< _Matrix_TypeA_ >::DiagSizeAtCompileTime;
 
-  Eigen::JacobiSVD< _Matrix_TypeA_ > svd = a.jacobiSvd(Eigen::ComputeThinU | Eigen::ComputeThinV);
+  Eigen::JacobiSVD< Eigen::Matrix<typename _Matrix_TypeA_::Scalar, Eigen::Dynamic, Eigen::Dynamic> > svd(a, Eigen::ComputeThinU | Eigen::ComputeThinV);
```

Comment:
This looks like a bad idea to me: it'll cause an implicit cast to a dynamic matrix, including a dynamic memory allocation. 
What was the reason?

Reply:
I just recently updated this method to support fixed size matrices. Maybe this dates back to when only dynamic ones were used...?

Reply:
the code as it was implemented did not work because thinU and thinV are only supported if the matrix colums are dynamic size:
https://eigen.tuxfamily.org/dox/classEigen_1_1JacobiSVD.html

I do not care much about the exact implementation as long as the unit test works ;)

Reply:
Did you try out the latest master? It works for me and the unit tests :)

Reply:
I did and it failed. I understand that it might work depending on compiler and Eigen version. But I do not think it is good practice to violate the API restrictions. I quote here:
"Thin unitaries are only available if your matrix type has a Dynamic number of columns (for example MatrixXf). They also are not available with the (non-default) FullPivHouseholderQR preconditioner."

Reply:
Sorry, @neunertm , for overlooking the commit message in the first place. I agree that it is a bad idea to rely on unspecified behavior. What about adding a comment with the URL, so this reason does not get forgotten? (maybe including a todo to poll the specs; ).

Reply:
Oops, I overlooked that. If we stick with this approach then as mentioned in my previous comment, we should remove leftCols(sigmathin.cols()), topRows(sigmaThin.rows()) in line 92.